arches/app/etl_modules/import_single_csv.py

```diff
@@ -70,15 +73,44 @@ def read(self, request):
         If the loadid already exsists also returns the load_details
         """
 
-        file = request.FILES.get("file")
-        csvfile = file.read().decode("utf-8")
-        reader = csv.reader(io.StringIO(csvfile))
-        data = {"csv": [line for line in reader]}
-        with connection.cursor() as cursor:
-            cursor.execute("""SELECT load_details FROM load_event WHERE loadid = %s""", [self.loadid])
-            row = cursor.fetchall()
-        if len(row) > 0:
-            data["config"] = row[0][0]
+        content = request.FILES.get("file")
+        temp_dir = os.path.join(settings.APP_ROOT, "tmp", self.loadid)
+        try:
+            shutil.rmtree(temp_dir)
+        except (FileNotFoundError):
+            pass
+        os.mkdir(temp_dir, 0o770)
+
+        csv_file_name = None
+        if content.content_type == "text/csv":
+            csv_file_name = content.name
+            csv_file_path = os.path.join(temp_dir, csv_file_name)
+            # maybe we can do this:
+            # default_storage.save(temp_dir, content)
+            with open(csv_file_path, "wb+") as destination:
+                for chunk in content.chunks():
+                    destination.write(chunk)
+        elif content.content_type == "application/zip":
+            with zipfile.ZipFile(content, "r") as zip_ref:
+                zip_ref.extractall(temp_dir)
+                files = zip_ref.infolist()
+                for file in files:
+                    if not file.filename.startswith("__MACOSX") and file.filename.endswith(".csv"):
```

Comment:
The code block is updated to use default_storage, so it is resolved.
A mac seems to create __MACOSX with all the files (metadata file?) the when zip files, so it may still be necessary to check.